File: src/hey/ReverseInteger.py
# problem: Reverse Integer
# author: everpuck
# date: 2/14/2019


class Solution:
    def reverse(self, x: 'int') -> 'int':
        if x == 0:
            return x
        xlist = list(str(x).strip('0'))
        if x < 0:
            xlist  = xlist  [1:]
        for i in range(len(xlist)//2):
            xlist[i], xlist[-(i + 1)] = xlist[-(i + 1)], xlist[i]
        
        ret = int("-" + "".join(xlist) if x < 0 else "".join(xlist))
        if ret < -2**31 or ret > 2**31 -1:
            return 0
        return ret


if __name__ == '__main__':
    x = 0
    solution = Solution()
    print(solution.reverse(x))

File: src/hey/ReverseInteger.java
/*
    problem: Reverse Integer
    author: everpuck
    date: 2/14/2019
*/


class Solution {
    public int reverse(int x) {
        // StringBuilder ret = new StringBuilder (Integer.toString(x));
        StringBuffer ret = new StringBuffer  (Integer.toString(x));
        ret.reverse();
        int iret = 0;

        try
        {
            if (x < 0) {
                iret = -1 * Integer.parseInt(ret.substring(0, ret.length()-1));
            }
            else {
                iret = Integer.parseInt(ret.toString());
            }
        }
        catch (Exception e)
        {
            return 0;
        }
        return iret;
    }
}

public class ReverseInteger {
    public static void main(String[] args) {
        // int x = 2147483647;
        int x = -10200;
        Solution solution = new Solution();
        int ret = solution.reverse(x);
        System.out.println(ret);
    }
}